### ISSUE TITLE
Any gus succeed to update moving_mean and moving_variance?

### ISSUE THREAD

**Issue by dragen1860** [cbfinn/maml#33]
Dear all:
  I try to use the moving version of batch norm:
```python
			# # add batch_norm ops before meta_op
			update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
			with tf.control_dependencies(update_ops):
				# update theta
				self.meta_op = optimizer.apply_gradients(gvs)
```
and set the flags of batch_normalization properly:
```python
x = tf.layers.batch_normalization(x, training=training, name=scope + '_bn', reuse=tf.AUTO_REUSE, fused=True)
```
However, it has errors when runing:
```shell
'gradients/truediv_7_grad/Neg' has inputs from different frames. The input 'map_fn/while/MAML_39/conv3_bn/AssignMovingAvg_1' is in frame 'map_fn/while/while_context'. The input 'Sum_5' is in frame ''.
```

I search google and can not find effective solutions yet. 
Anyone succeed to use the moving version of batch_norm?? please help me or share part of your code.

**Comment by dragen1860** [comment]
I found the solution.
Put `update_ops` in `meta_task` function to make `update_ops` in `while_loop`.
and set `training=your_train_or_test_flag` for `tf.layers.batch_normallization`.
Do NOT forget to use `tf.train.Saver(tf.global_variables())` to save `moving_mean, moving_variance`, otherwise the `moving_mean, moving_variance` will not be saved into checkpoint.
```python
			update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
			with tf.control_dependencies(update_ops):
				support_loss = tf.identity(support_loss)
```
